pkg/build/api/validation/validation.go

```diff
@@ -186,6 +186,8 @@ func validateSource(input *buildapi.BuildSource) fielderrors.ValidationErrorList
 		}
 	case "":
 		allErrs = append(allErrs, fielderrors.NewFieldRequired("type"))
+	default:
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("type", input.Type, fmt.Sprintf("source type must be one of Git, Dockerfile, or Binary")))
```

Comment:
@bparees aren't we introducing a problem we had before here (excessive validation in trigger types)?
@csrwng [mentioned it in the BZ](https://bugzilla.redhat.com/show_bug.cgi?id=1279288#c1) and I agree with him on not adding validation here.

Adding a new source type will make old servers throw validation error instead of gracefully ignoring the fields it doesn't know about.